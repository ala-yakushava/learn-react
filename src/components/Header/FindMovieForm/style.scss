@import "../../../styles/theme";

.FindMovieForm {
  display: grid;
  grid-template-rows: 2;
  grid-template-columns: 1fr min-content;
  column-gap: 15px;

  &_title {
    grid-column: 1/3;
    grid-row: 1/2;
  }

  &_input {
    grid-column: 1/2;
    grid-row: 2/3;
  }
}
