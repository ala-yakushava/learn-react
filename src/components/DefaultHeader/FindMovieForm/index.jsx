import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { useFormik } from 'formik';

import './style.scss';
import Heading from '../../Heading';
import TextInput from '../../TextInput';
import Button from '../../Button';

const FindMovieForm = ({ className }) => {
  const formik = useFormik({
    initialValues: { text: '' },
    onSubmit: (values) => {
      console.log(values);
    },
  });

  const FormClass = cn('FindMovieForm', className);

  const {
    values, isSubmitting, dirty, isValid, handleSubmit, handleChange,
  } = formik;

  return (
    <form className={FormClass} onSubmit={handleSubmit}>
      <Heading className="FindMovieForm_title">Find your movie</Heading>
      <TextInput
        name="text"
        className="FindMovieForm_input"
        placeholder="What do you want to watch?"
        value={values.text}
        required
        onChange={handleChange}
      />
      <Button type="submit" mode="primary" disabled={isSubmitting || !dirty || !isValid}>Search</Button>
    </form>
  );
};

FindMovieForm.propTypes = {
  className: PropTypes.string,
};

FindMovieForm.defaultProps = {
  className: '',
};

export default FindMovieForm;
