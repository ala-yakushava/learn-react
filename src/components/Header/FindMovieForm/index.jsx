import React from 'react';

import './style.scss';
import Heading from '../../Heading';
import TextInput from '../../TextInput';
import Button from '../../Button';

class FindMovieForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      text: '',
    };
  }

  handleInputChange = () => (evt) => {
    const { target: { name, value } } = evt;
    this.setState({
      [name]: value,
    });
  }

  handleSubmit = (evt) => {
    console.log(this.state);
    evt.preventDefault();
  }

  render() {
    return (
      <form className="FindMovieForm" onSubmit={this.handleSubmit}>
        <Heading className="FindMovieForm_title">Find your movie</Heading>
        <TextInput
          name="text"
          className="FindMovieForm_input"
          placeholder="What do you want to watch?"
          value={this.state.text}
          isRequired={true}
          onChange={this.handleInputChange(this.state.text)}
        />
        <Button type="submit" mode="primary">Search</Button>
      </form>
    );
  }
}

export default FindMovieForm;
