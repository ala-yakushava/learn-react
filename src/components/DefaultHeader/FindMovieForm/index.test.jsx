import React from 'react';
import renderer from 'react-test-renderer';

import FindMovieForm from './index';

describe('Components - FindMovieForm', () => {
  test('should match snapshot', () => {
    const tree = renderer
      .create(<FindMovieForm />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
