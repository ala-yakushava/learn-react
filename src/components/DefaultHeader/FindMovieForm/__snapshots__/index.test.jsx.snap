// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - FindMovieForm should match snapshot 1`] = `
<form
  className="FindMovieForm"
  onSubmit={[Function]}
>
  <div
    className="Heading FindMovieForm_title"
  >
    Find your movie
  </div>
  <div
    className="TextInput"
  >
    <input
      className="TextInput_field"
      id="text"
      name="text"
      onChange={[Function]}
      placeholder="What do you want to watch?"
      readOnly={false}
      required={true}
      type="text"
      value=""
    />
  </div>
  <button
    className="Button Button--primary"
    disabled={true}
    onClick={[Function]}
    type="submit"
  >
    <span
      className="Button_text"
    >
      Search
    </span>
  </button>
</form>
`;
